### PR TITLE
add reset metadata button to settings page (close #515)

### LINKED ISSUE #515: add a reset metadata button on console
This would be under settings page, below import/export.

### PR DESCRIPTION
**Pull request by paranoidsp**
### Description

What component does this PR affect? 

- [ ] Server
- [X] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] Community content

### Checklist:

- [X] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

**Comment by paranoidsp** [comment]
Waiting on a small text change before merging.

File: console/src/components/Services/Data/Metadata/ResetMetadata.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                ...this.props.dataHeaders,
+              },
+              body: JSON.stringify(requestBody),
+            };
+            fetch(url, options).then(response => {
+              response.json().then(data => {
+                if (response.ok) {
+                  this.setState({ isResetting: false });
+                  this.props.dispatch(
+                    showSuccessNotification('Metadata reset successfully!')
```
Comment (karthikvt26) [inline console/src/components/Services/Data/Metadata/ResetMetadata.js]: Add `are you sure` pop up as suggested by sandip

File: console/src/components/Services/Data/Metadata/ResetMetadata.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const metaDataStyles = require('./Metadata.scss');
+    return (
+      <div className={metaDataStyles.display_inline}>
+        <button
+          data-test="data-reset-metadata"
+          className={styles.default_button + ' ' + metaDataStyles.margin_right}
+          onClick={e => {
+            e.preventDefault();
+            const a = prompt(
+              'Are you sure you absolutely sure?\n This action cannot be undone. This will permanently reset GraphQL Engine\'s configuration and you will need to start from scratch. Please type "Reset" (in caps, without quotes) to confirm. '
```
Comment (karthikvt26) [inline console/src/components/Services/Data/Metadata/ResetMetadata.js]: @paranoidsp typo?

**Comment by paranoidsp** [comment]
@karthikvt26 Fixed both issues, needs review.

**Comment by shahidhk** [comment]
If I enter something other than `RESET`, an error appears with a details button, which on clicking shows this:
image

can we remove the details button?

**Comment by paranoidsp** [comment]
@shahidhk @karthikvt26 Fixed, check now?

**Pull request merged**